frontend/src/components/visual-editor/v2/Diagrams.tsx

```diff
@@ -380,48 +380,52 @@ const Diagrams = () => {
       } else {
         // Block Delete Case
         const ids = id.includes(",") ? id.split(",") : [id];
-        const deletePromises = ids.map((id) => {
-          const block = getBlockFromCache(id);
 
-          return deleteBlock(id, {
+        for (const blockId of ids) {
```

Comment:
Thanks @IkbelTalebHssan 

Can we rather implement deleteMany() on the API side ?

Reply:
Thanks :pray: 
I made some updates can you please check again :pray: